### DISCUSSION TITLE
Building Service

### DISCUSSION THREAD

**Discussion by vsoch** [supercontainers/containers-wg#5] [Ideas]
This is a discussion thread for the [Building Service](https://supercontainers.github.io/containers-wg/ideas/building-service/) idea. You can:

 - vote this discussion up to express interest in the idea
 - add comments, feedback, or ideas for discussion in this thread.

**Comment by dtrudg** [comment]
As an (unfortunate) but timely practical matter... any build service specification that would ultimately be implemented in a publicly accessible service needs to be flexible enough to accommodate usage limits and protections, in a way that doesn't require breaking the spec. 

A publicly accessible free service implementing a spec is a great way to get traction, but funding public free build services, and defending against cryptocurrency mining abuse is leading to continued loss of free public container and CI services at present.

**Reply by vsoch** [reply]
I think that flexibility wouldn't be hard to do, and the second problem you mentioned I don't think should deter us from pursuing any idea that we think is worthwhile. As someone that ran a "free build service" for five years, I will say that it's possible, and I suspect we just need another solution to prevent cryptocurrency mining.
